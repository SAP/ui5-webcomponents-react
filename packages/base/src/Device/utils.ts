import { getOS, supportTouch } from '@ui5/webcomponents-base/dist/Device.js';
import { supportMatchMedia, supportOrientation } from './Support';

export const getWindowSize = () => {
  return [window.innerWidth, window.innerHeight];
};

export const changeRootCSSClass = (sClassName, bRemove, sPrefix?) => {
  const oRoot = document.documentElement;
  if (oRoot.className.length === 0) {
    if (!bRemove) {
      oRoot.className = sClassName;
    }
  } else {
    const aCurrentClasses = oRoot.className.split(' ');
    let sNewClasses = '';
    aCurrentClasses.forEach((currentClass) => {
      // eslint-disable-next-line @typescript-eslint/prefer-string-starts-ends-with
      if ((sPrefix && currentClass.indexOf(sPrefix) !== 0) || (!sPrefix && currentClass !== sClassName)) {
        sNewClasses = `${sNewClasses + currentClass} `;
      }
    });
    if (!bRemove) {
      sNewClasses += sClassName;
    }
    oRoot.className = sNewClasses;
  }
};

export const getQuery = (from: number, to: number, unit = 'px') => {
  let q = 'all';
  if (from > 0) {
    q = `${q} and (min-width:${from}${unit})`;
  }
  if (to > 0) {
    q = `${q} and (max-width:${to}${unit})`;
  }
  return q;
};

export const convertToPx = (val, unit?) => {
  if (unit === 'em' || unit === 'rem') {
    const s = window.getComputedStyle ?? ((element: any) => element.currentStyle);
    const x = s(document.documentElement).fontSize;
    const f = x && x.indexOf('px') >= 0 ? parseFloat(x) : 16;
    return val * f;
  }
  return val;
};

export const matchLegacyBySize = (from, to, unit, size) => {
  from = convertToPx(from, unit);
  to = convertToPx(to, unit);

  const width = size[0];
  const a = from < 0 || from <= width;
  const b = to < 0 || width <= to;
  return a && b;
};

export const isLandscape = (bFromOrientationChange: boolean, orientation, bKeyboardOpen: boolean) => {
  if (supportTouch() && supportOrientation() && getOS().android) {
    // if on screen keyboard is open and the call of this method is from orientation change listener,
    // reverse the last value. this is because when keyboard opens on android device, the height can be less
    // than the width even in portrait mode.
    if (bKeyboardOpen && bFromOrientationChange) {
      return !orientation.landscape;
    }
    if (bKeyboardOpen) {
      // when keyboard opens, the last orientation change value will be returned.
      return orientation.landscape;
    }
  } else if (supportMatchMedia() && supportOrientation()) {
    // most desktop browsers and windows phone/tablet which not support orientationchange
    return !!window.matchMedia('(orientation: landscape)').matches;
  }
  // otherwise compare the width and height of window
  const size = getWindowSize();
  return size[0] > size[1];
};
