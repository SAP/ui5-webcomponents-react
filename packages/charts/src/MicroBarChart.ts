import { deprecationNotice } from '@ui5/webcomponents-react-base';
import { MicroBarChart } from './components/MicroBarChart/MicroBarChart';

deprecationNotice(
  'MicroBarChart',
  `importing from '@ui5/webcomponents-react-charts/dist/MicroBarChart' is deprecated. Please use "import { MicroBarChart } from '@ui5/webcomponents-react-charts';" instead.`
);
export { MicroBarChart };
