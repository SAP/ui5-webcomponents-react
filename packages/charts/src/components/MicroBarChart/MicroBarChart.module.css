.container {
  display: flex;
  flex-direction: column;
  overflow: hidden;
  font-family: var(--sapFontFamily);
  font-weight: normal;
  width: 100%;
  height: 100%;
  justify-content: space-around;
}

.barContainer {
  cursor: auto;
}

.barContainerActive {
  cursor: pointer;

  &:active {
    opacity: 0.3 !important;
  }
}

.labelContainer {
  display: flex;
  justify-content: space-between;
}

.valueContainer {
  display: flex;
  background-color: var(--sapContent_Placeholderloading_Background);
}

.valueBar {
  height: 100%;
}

.label {
  color: var(--sapContent_LabelColor);
  display: block;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
  font-size: var(--sapFontSmallSize);
  max-width: 70%;
}

.text {
  padding-inline-start: 6px;
  display: inline-block;
  overflow: hidden;
  font-size: var(--sapFontSmallSize);
  box-sizing: border-box;
  white-space: nowrap;
  text-overflow: ellipsis;
  color: var(--sapTextColor);
  text-align: right;
}
