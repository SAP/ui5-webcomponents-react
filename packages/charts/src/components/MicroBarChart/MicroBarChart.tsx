import { createUseStyles } from 'react-jss';
import { StyleClassHelper } from '@ui5/webcomponents-react-base/dist/StyleClassHelper';
import { ThemingParameters } from '@ui5/webcomponents-react-base/dist/ThemingParameters';
import { useConsolidatedRef } from '@ui5/webcomponents-react-base/dist/useConsolidatedRef';
import { usePassThroughHtmlProps } from '@ui5/webcomponents-react-base/dist/usePassThroughHtmlProps';
import { enrichEventWithDetails } from '@ui5/webcomponents-react-base/dist/Utils';
import { BarChartPlaceholder } from '@ui5/webcomponents-react-charts/dist/BarChartPlaceholder';
import { ChartContainer } from '@ui5/webcomponents-react-charts/dist/components/ChartContainer';
import React, { CSSProperties, FC, forwardRef, Ref, useCallback, useMemo } from 'react';
import { getValueByDataKey } from 'recharts/lib/util/ChartUtils';
import { IChartBaseProps } from '../../interfaces/IChartBaseProps';
import { IChartDimension } from '../../interfaces/IChartDimension';
import { IChartMeasure } from '../../interfaces/IChartMeasure';
import { defaultFormatter } from '../../internal/defaults';

interface MeasureConfig extends Omit<IChartMeasure, 'color'> {
  /**
   * array of any valid CSS Color or CSS Variable. Defaults to the `sapChart_OrderedColor_` colors
   */
  colors?: CSSProperties['color'][];
  /**
   * Bar Opacity
   * @default 1
   */
  opacity?: number;
  /**
   * Bar Width
   * @default auto
   */
  width?: number;
}

export interface MicroBarChartProps
  extends Omit<
    IChartBaseProps,
    'noLegend' | 'onLegendClick' | 'noAnimation' | 'chartConfig' | 'children' | 'tooltipConfig'
  > {
  /**
   * A object which contains the configuration of the dimension.
   *
   * #### Required Properties
   * - `accessor`: string containing the path to the dataset key the dimension should display. Supports object structures by using <code>'parent.child'</code>.
   *   Can also be a getter.
   *
   * #### Optional Properties
   * - `formatter`: function will be called for each data label and allows you to format it according to your needs
   *
   */
  dimension: IChartDimension;
  /**
   * An array of config objects. Each object is defining one bar in the chart.
   *
   * #### Required properties
   * - `accessor`: string containing the path to the dataset key this bar should display. Supports object structures by using <code>'parent.child'</code>.
   * Can also be a getter.
   *
   * #### Optional properties
   *
   * - `formatter`: function will be called for each data label and allows you to format it according to your needs
   * - `colors`: array of any valid CSS Color or CSS Variable. Defaults to the `sapChart_OrderedColor_` colors
   * - `width`: bar width in pixel, defaults to `4`
   * - `opacity`: bar opacity, defaults to `1`
   * - `hideDataLabel`: flag whether the data labels should be hidden in the chart for this column.
   * - `DataLabel`: a custom component to be used for the data label
   */
  measure: MeasureConfig;
  /**
   * An optional number for the maxValue of the valueBar.
   * Default is the highest number of the corresponding accessor in the dataset.
   */
  maxValue?: number;
}

const resolveColor = (index: number, color = null) => {
  if (color) {
    return ThemingParameters[color] ?? color;
  }
  return ThemingParameters[`sapChart_Sequence_${(index % 11) + 1}`];
};

const MicroBarChartStyles = {
  container: {
    display: 'flex',
    flexDirection: 'column',
    overflow: 'hidden',
    fontFamily: ThemingParameters.sapFontFamily,
    fontWeight: 'normal',
    width: '100%',
    height: '100%',
    justifyContent: 'space-around'
  },
  barContainer: {
    cursor: 'auto'
  },
  barContainerActive: {
    '&:active': { opacity: '0.3 !important' },
    cursor: 'pointer'
  },
  labelContainer: {
    display: 'flex',
    justifyContent: 'space-between'
  },
  valueContainer: {
    display: 'flex',
    backgroundColor: ThemingParameters.sapContent_Placeholderloading_Background
  },
  valueBar: { height: '100%' },
  label: {
    color: ThemingParameters.sapContent_LabelColor,
    display: 'block',
    overflow: 'hidden',
    whiteSpace: 'nowrap',
    textOverflow: 'ellipsis',
    fontSize: ThemingParameters.sapFontSmallSize,
    maxWidth: '70%'
  },
  text: {
    paddingLeft: '6px',
    display: 'inline-block',
    overflow: 'hidden',
    fontSize: ThemingParameters.sapFontSmallSize,
    boxSizing: 'border-box',

    whiteSpace: 'nowrap',
    textOverflow: 'ellipsis',
    color: ThemingParameters.sapTextColor,
    textAlign: 'right'
  }
};

const useStyles = createUseStyles(MicroBarChartStyles, { name: 'MicroBarChart' });

/**
 * The `MicroBarChart` compares different values of the same category to each other by displaying them in a compact way.
 */
const MicroBarChart: FC<MicroBarChartProps> = forwardRef((props: MicroBarChartProps, ref: Ref<HTMLDivElement>) => {
  const { loading, dataset, onDataPointClick, style, className, tooltip, slot, ChartPlaceholder } = props;
  const classes = useStyles();

  const dimension = useMemo<IChartDimension>(
    () => ({
      formatter: defaultFormatter,
      ...props.dimension
    }),
    [props.dimension]
  );
  const measure: MeasureConfig = useMemo(
    () => ({
      formatter: defaultFormatter,
      ...props.measure
    }),
    [props.measure]
  );

  const maxValue = useMemo(() => {
    if (dataset) {
      const maxDatasetValue = Math.max(...dataset?.map((item) => getValueByDataKey(item, measure.accessor)));
      return props.maxValue ?? maxDatasetValue;
    }
    return 0;
  }, [dataset, measure, props.maxValue]);
  const chartRef = useConsolidatedRef<any>(ref);

  const barHeight = measure?.width ? `${measure.width}px` : '4px';

  const onBarClick = useCallback(
    (item, index) => (e) => {
      if (typeof onDataPointClick === 'function') {
        onDataPointClick(
          enrichEventWithDetails(e, {
            dataKey: measure.accessor,
            value: getValueByDataKey(item, measure.accessor),
            payload: item,
            dataIndex: index
          })
        );
      }
    },
    [measure.accessor, onDataPointClick]
  );
  const barContainerClasses = StyleClassHelper.of(classes.barContainer);
  if (onDataPointClick) {
    barContainerClasses.put(classes.barContainerActive);
  }
  const passThroughProps = usePassThroughHtmlProps(props, ['onDataPointClick', 'onLegendClick']);

  return (
    <ChartContainer
      dataset={dataset}
      loading={loading}
      Placeholder={ChartPlaceholder ?? BarChartPlaceholder}
      ref={chartRef}
      style={style}
      className={className}
      tooltip={tooltip}
      slot={slot}
      resizeDebounce={250}
      {...passThroughProps}
    >
      <div className={classes.container}>
        {dataset?.map((item, index) => {
          const dimensionValue = getValueByDataKey(item, dimension.accessor);
          const measureValue = getValueByDataKey(item, measure.accessor);
          const formattedDimension = dimension.formatter(dimensionValue);
          let formattedMeasure: any = '';
          if (!measure.hideDataLabel) {
            if (measure.DataLabel) {
              formattedMeasure = React.createElement(measure.DataLabel, {
                value: measureValue,
                config: measure,
                formattedValue: measure.formatter(measureValue)
              });
            } else {
              formattedMeasure = measure.formatter(measureValue);
            }
          }
          return (
            <div key={dimensionValue} className={barContainerClasses.className} onClick={onBarClick(item, index)}>
              <div className={classes.labelContainer}>
                <span className={classes.label} title={formattedDimension}>
                  {formattedDimension}
                </span>
                <span className={classes.text} title={formattedMeasure}>
                  {measure.hideDataLabel ? '' : formattedMeasure}
                </span>
              </div>
              <div
                className={classes.valueContainer}
                style={{
                  opacity: measure?.opacity ?? 1,
                  height: barHeight
                }}
              >
                <div
                  className={classes.valueBar}
                  style={{
                    width: `${(measureValue / maxValue) * 100}%`,
                    backgroundColor: resolveColor(index, measure?.colors?.[index])
                  }}
                />
              </div>
            </div>
          );
        })}
      </div>
    </ChartContainer>
  );
});

MicroBarChart.displayName = 'MicroBarChart';

export { MicroBarChart };
