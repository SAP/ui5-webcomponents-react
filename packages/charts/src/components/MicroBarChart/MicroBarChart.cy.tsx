import { MicroBarChart } from './MicroBarChart.js';
import { cypressPassThroughTestsFactory } from '@/cypress/support/utils';

const text1 = 'January';
const text2 = 'February';
const text3 = 'March';

const dataset = [
  { value: 10, label: text1 },
  { value: 100, label: text2 },
  { value: 70, label: text3 }
];

const dimension = {
  accessor: 'label'
};

const measure = {
  accessor: 'value'
};

describe('MicroBarChart', () => {
  it('Basic', () => {
    cy.mount(<MicroBarChart dataset={dataset} dimension={dimension} measure={measure} />);
    cy.get('.recharts-responsive-container').should('be.visible');
    cy.get('div[class^=MicroBarChart-valueBar]').should('have.length', 3);

    cy.findByText(text1).should('be.visible');
    cy.findByText(text2).should('be.visible');
    cy.findByText(text3).should('be.visible');
  });

  it('Dimension Formatter', () => {
    cy.mount(
      <MicroBarChart
        dataset={dataset}
        dimension={{ ...dimension, formatter: (label) => `${label} - formatted` }}
        measure={measure}
      />
    );
    cy.get('.recharts-responsive-container').should('be.visible');
    cy.get('div[class^=MicroBarChart-valueBar]').should('have.length', 3);

    cy.findByText(`${text1} - formatted`).should('be.visible');
    cy.findByText(`${text2} - formatted`).should('be.visible');
    cy.findByText(`${text3} - formatted`).should('be.visible');
  });

  it('click handlers', () => {
    const onClick = cy.spy().as('onClick');
    cy.mount(<MicroBarChart dataset={dataset} dimension={dimension} measure={measure} onDataPointClick={onClick} />);

    cy.findByText('January').click();
    cy.get('@onClick')
      .should('have.been.calledOnce')
      .and(
        'have.been.calledWith',
        Cypress.sinon.match({
          detail: Cypress.sinon.match({
            payload: dataset[0]
          })
        })
      );
  });

  it('Loading Placeholder', () => {
    cy.mount(<MicroBarChart dataset={[]} dimension={dimension} measure={measure} />);
    cy.get('.recharts-bar').should('not.exist');
    cy.contains('Loading...').should('exist');
  });

  cypressPassThroughTestsFactory(MicroBarChart, { dimension, measure });
});
