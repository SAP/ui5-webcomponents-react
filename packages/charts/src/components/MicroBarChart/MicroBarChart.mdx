import { ControlsWithNote, DocsHeader, Footer } from '@sb/components';
import { Canvas, Meta } from '@storybook/addon-docs';
import * as ComponentStories from './MicroBarChart.stories';

<Meta of={ComponentStories} />

<DocsHeader />

<br />

## Example

<Canvas of={ComponentStories.Default} />

## Properties

<ControlsWithNote of={ComponentStories.Default} />

<br />
<br />

## More Examples

### With Formatter

<Canvas of={ComponentStories.WithFormatter} />

### Loading Placeholder

<Canvas of={ComponentStories.LoadingPlaceholder} />

<Footer />
