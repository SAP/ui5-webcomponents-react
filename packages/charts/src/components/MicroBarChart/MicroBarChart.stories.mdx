import { ArgsTable, Canvas, Meta, Story } from '@storybook/addon-docs';
import { MicroBarChart } from './MicroBarChart';
import { DocsHeader } from '@docs/DocsHeader';
import { Footer } from '@docs/Footer';

<Meta
  title="Charts /  MicroBarChart"
  component={MicroBarChart}
  argTypes={{
    dataset: { control: { disable: true } }
  }}
  args={{
    dataset: [
      {
        name: 'January',
        data: 100
      },
      {
        name: 'February',
        data: 300
      },
      {
        name: 'March',
        data: 530
      },
      {
        name: 'April',
        data: 200
      }
    ]
  }}
/>

<DocsHeader />

<br />

## Example

<Canvas>
  <Story name="Default">
    {(args) => (
      <MicroBarChart
        dataset={args.dataset}
        loading={args.loading}
        style={{ width: '100%' }}
        dimension={{
          accessor: 'name'
        }}
        measure={{
          accessor: 'data'
        }}
      />
    )}
  </Story>
</Canvas>

## Properties

<ArgsTable story="." />

<br />
<br />

## More Examples

### With Custom Props

<Canvas>
  <Story name="With custom props">
    {(args) => (
      <MicroBarChart
        onDataPointClick={args.onDataPointClick}
        dataset={args.dataset}
        style={{ width: '50%' }}
        loading={args.loading}
        dimension={{
          accessor: 'name'
        }}
        measure={{
          accessor: 'data',
          colors: ['#f0ab00', 'var(--sapHighlightColor)'],
          formatter: (d) => (d > 200 ? 'over 200' : 'lower'),
          width: 16
        }}
      />
    )}
  </Story>
</Canvas>

### Loading Placeholder

<Canvas>
  <Story name="Loading Placeholder">
    <MicroBarChart
      dataset={[]}
      dimension={{
        accessor: 'name'
      }}
      measure={{
        accessor: 'data'
      }}
      style={{ width: '20%' }}
    />
  </Story>
</Canvas>

<Footer />
