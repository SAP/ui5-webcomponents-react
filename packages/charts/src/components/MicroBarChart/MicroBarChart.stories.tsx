import { Meta, StoryObj } from '@storybook/react';
import { MicroBarChart } from './MicroBarChart.js';

const meta = {
  title: 'MicroBarChart',
  component: MicroBarChart,
  args: {
    dimension: {
      accessor: 'name'
    },
    measure: { accessor: 'data' },
    dataset: [
      {
        name: 'January',
        data: 100
      },
      {
        name: 'February',
        data: 300
      },
      {
        name: 'March',
        data: 530
      },
      {
        name: 'April',
        data: 200
      }
    ]
  },
  argTypes: {
    dataset: { control: { disable: true } }
  }
} satisfies Meta<typeof MicroBarChart>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {};

export const WithFormatter: Story = {
  args: {
    measure: {
      accessor: 'data',
      colors: ['#f0ab00', 'var(--sapHighlightColor)'],
      formatter: (d) => (d > 200 ? 'over 200' : 'lower'),
      width: 16
    }
  }
};

export const LoadingPlaceholder: Story = {
  args: {
    dataset: []
  }
};
