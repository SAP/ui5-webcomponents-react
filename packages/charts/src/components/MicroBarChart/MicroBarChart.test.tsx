import * as React from 'react';
import { render, screen, fireEvent } from '@shared/tests/index';
import { MicroBarChart } from './MicroBarChart';
import { createChartRenderTest, createLoadingPlaceholderTest, createOnClickChartTest } from '@shared/tests/chartUtils';
import { createCustomPropsTest } from '@shared/tests/utils';

const text1 = 'January';
const text2 = 'February';
const text3 = 'March';

const dataset = [
  { value: 10, label: text1 },
  { value: 100, label: text2 },
  { value: 70, label: text3 }
];

describe('Micro Bar Chart', () => {
  it('Render with default Props', () => {
    const utils = render(
      <MicroBarChart
        dimension={{
          accessor: 'label'
        }}
        measure={{
          accessor: 'value'
        }}
        dataset={dataset}
      />
    );
    expect(screen.getByText(text1).textContent).toEqual(text1);
    expect(screen.getByText(text2).textContent).toEqual(text2);
    expect(screen.getByText(text3).textContent).toEqual(text3);

    expect(utils.asFragment()).toMatchSnapshot();
  });

  it('With formatted dimension', () => {
    render(
      <MicroBarChart
        dimension={{
          accessor: 'label',
          formatter: (label) => `${label} - formatted`
        }}
        measure={{
          accessor: 'value'
        }}
        dataset={dataset}
      />
    );
    expect(screen.getByText('January - formatted').textContent).toEqual('January - formatted');
    expect(screen.getByText('February - formatted').textContent).toEqual('February - formatted');
  });

  it('With custom colors', () => {
    expect(
      render(
        <MicroBarChart
          dimension={{
            accessor: 'label',
            formatter: (label) => `${label} - formatted`
          }}
          measure={{
            accessor: 'value',
            colors: ['black', 'yellow']
          }}
          dataset={dataset}
        />
      ).asFragment()
    ).toMatchSnapshot();
  });

  it('Bar click', async () => {
    let internalDataset = dataset;

    const { getByText, rerender } = render(
      <MicroBarChart
        dimension={{
          accessor: 'label'
        }}
        onDataPointClick={() =>
          (internalDataset = [
            { value: 10, label: 'January' },
            { value: 100, label: 'February - clicked' },
            { value: 70, label: 'March' }
          ])
        }
        measure={{
          accessor: 'value'
        }}
        dataset={internalDataset}
      />
    );
    fireEvent.click(getByText('February'));

    rerender(
      <MicroBarChart
        dimension={{
          accessor: 'label'
        }}
        measure={{
          accessor: 'value'
        }}
        dataset={internalDataset}
      />
    );
    expect(screen.getByText('February - clicked').textContent).toEqual('February - clicked');
  });

  createChartRenderTest(MicroBarChart, {
    dataset,
    measure: {
      accessor: 'value'
    },
    dimension: {
      accessor: 'label'
    }
  });

  createOnClickChartTest(MicroBarChart, {
    dataset,
    noLegend: true,
    measure: {
      accessor: 'value'
    },
    dimension: {
      accessor: 'label'
    }
  });

  createLoadingPlaceholderTest(MicroBarChart, { dimension: [], measure: [] });

  createCustomPropsTest(MicroBarChart, { dimension: [], measure: [] });
});
