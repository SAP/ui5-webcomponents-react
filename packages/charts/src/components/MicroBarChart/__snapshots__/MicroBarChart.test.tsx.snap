// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Micro Bar Chart Check onClick events 1`] = `
<DocumentFragment>
  <div
    style="font-family: var(--sapFontFamily); width: 100%; height: 400px; position: relative;"
  >
    <div
      class="recharts-responsive-container"
      style="width: 400px; height: 400px;"
    >
      <div
        class="MicroBarChart-container"
        height="400"
        width="400"
      >
        <div
          class="MicroBarChart-barContainer"
        >
          <div
            class="MicroBarChart-labelContainer"
          >
            <span
              class="MicroBarChart-label"
              title="January"
            >
              January
            </span>
            <span
              class="MicroBarChart-text"
            />
          </div>
          <div
            class="MicroBarChart-valueContainer"
            style="opacity: 1; height: 4px;"
          >
            <div
              class="MicroBarChart-valueBar"
            />
          </div>
        </div>
        <div
          class="MicroBarChart-barContainer"
        >
          <div
            class="MicroBarChart-labelContainer"
          >
            <span
              class="MicroBarChart-label"
              title="February"
            >
              February
            </span>
            <span
              class="MicroBarChart-text"
            />
          </div>
          <div
            class="MicroBarChart-valueContainer"
            style="opacity: 1; height: 4px;"
          >
            <div
              class="MicroBarChart-valueBar"
            />
          </div>
        </div>
        <div
          class="MicroBarChart-barContainer"
        >
          <div
            class="MicroBarChart-labelContainer"
          >
            <span
              class="MicroBarChart-label"
              title="March"
            >
              March
            </span>
            <span
              class="MicroBarChart-text"
            />
          </div>
          <div
            class="MicroBarChart-valueContainer"
            style="opacity: 1; height: 4px;"
          >
            <div
              class="MicroBarChart-valueBar"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Micro Bar Chart Loading placeholder 1`] = `
<DocumentFragment>
  <div
    style="font-family: var(--sapFontFamily); width: 100%; height: 400px; position: relative;"
  >
    <svg
      aria-labelledby="TITLE_ARIA_ID"
      height="100%"
      preserveAspectRatio="xMidYMid meet"
      role="img"
      viewBox="0 0 165 145"
      width="100%"
    >
      <title
        id="TITLE_ARIA_ID"
      >
        Loading...
      </title>
      <rect
        clip-path="STYLE-URL"
        height="100%"
        role="presentation"
        style="fill: STYLE-URL;"
        width="100%"
        x="0"
        y="0"
      />
      <defs>
        <clippath
          id="DIFF_ID"
        >
          <rect
            height="135"
            width="1"
            x="20"
            y="10"
          />
          <rect
            height="15"
            width="85"
            x="20"
            y="20"
          />
          <rect
            height="15"
            width="117"
            x="20"
            y="40"
          />
          <rect
            height="15"
            width="67"
            x="20"
            y="60"
          />
          <rect
            height="15"
            width="75"
            x="20"
            y="80"
          />
          <rect
            height="15"
            width="125"
            x="20"
            y="100"
          />
          <rect
            height="15"
            width="60"
            x="20"
            y="120"
          />
        </clippath>
        <lineargradient
          id="DIFF_ID"
        >
          <stop
            offset="0%"
            stop-color="var(--sapContent_ImagePlaceholderBackground)"
            stop-opacity="var(--sapContent_DisabledOpacity)"
          >
            <animate
              attributeName="offset"
              dur="2s"
              keyTimes="0; 0.25; 1"
              repeatCount="indefinite"
              values="-2; -2; 1"
            />
          </stop>
          <stop
            offset="50%"
            stop-color="var(--sapContent_ImagePlaceholderForegroundColor)"
            stop-opacity="1"
          >
            <animate
              attributeName="offset"
              dur="2s"
              keyTimes="0; 0.25; 1"
              repeatCount="indefinite"
              values="-1; -1; 2"
            />
          </stop>
          <stop
            offset="100%"
            stop-color="var(--sapContent_ImagePlaceholderBackground)"
            stop-opacity="var(--sapContent_DisabledOpacity)"
          >
            <animate
              attributeName="offset"
              dur="2s"
              keyTimes="0; 0.25; 1"
              repeatCount="indefinite"
              values="0; 0; 3"
            />
          </stop>
        </lineargradient>
      </defs>
    </svg>
  </div>
</DocumentFragment>
`;

exports[`Micro Bar Chart Render chart with data 1`] = `
<DocumentFragment>
  <div
    style="font-family: var(--sapFontFamily); width: 100%; height: 400px; position: relative;"
  >
    <div
      class="recharts-responsive-container"
      style="width: 400px; height: 400px;"
    >
      <div
        class="MicroBarChart-container"
        height="400"
        width="400"
      >
        <div
          class="MicroBarChart-barContainer"
        >
          <div
            class="MicroBarChart-labelContainer"
          >
            <span
              class="MicroBarChart-label"
              title="January"
            >
              January
            </span>
            <span
              class="MicroBarChart-text"
            />
          </div>
          <div
            class="MicroBarChart-valueContainer"
            style="opacity: 1; height: 4px;"
          >
            <div
              class="MicroBarChart-valueBar"
            />
          </div>
        </div>
        <div
          class="MicroBarChart-barContainer"
        >
          <div
            class="MicroBarChart-labelContainer"
          >
            <span
              class="MicroBarChart-label"
              title="February"
            >
              February
            </span>
            <span
              class="MicroBarChart-text"
            />
          </div>
          <div
            class="MicroBarChart-valueContainer"
            style="opacity: 1; height: 4px;"
          >
            <div
              class="MicroBarChart-valueBar"
            />
          </div>
        </div>
        <div
          class="MicroBarChart-barContainer"
        >
          <div
            class="MicroBarChart-labelContainer"
          >
            <span
              class="MicroBarChart-label"
              title="March"
            >
              March
            </span>
            <span
              class="MicroBarChart-text"
            />
          </div>
          <div
            class="MicroBarChart-valueContainer"
            style="opacity: 1; height: 4px;"
          >
            <div
              class="MicroBarChart-valueBar"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Micro Bar Chart Render with default Props 1`] = `
<DocumentFragment>
  <div
    style="font-family: var(--sapFontFamily); width: 100%; height: 400px; position: relative;"
  >
    <div
      class="recharts-responsive-container"
      style="width: 400px; height: 400px;"
    >
      <div
        class="MicroBarChart-container"
        height="400"
        width="400"
      >
        <div
          class="MicroBarChart-barContainer"
        >
          <div
            class="MicroBarChart-labelContainer"
          >
            <span
              class="MicroBarChart-label"
              title="January"
            >
              January
            </span>
            <span
              class="MicroBarChart-text"
              title="10"
            >
              10
            </span>
          </div>
          <div
            class="MicroBarChart-valueContainer"
            style="opacity: 1; height: 4px;"
          >
            <div
              class="MicroBarChart-valueBar"
              style="width: 10%;"
            />
          </div>
        </div>
        <div
          class="MicroBarChart-barContainer"
        >
          <div
            class="MicroBarChart-labelContainer"
          >
            <span
              class="MicroBarChart-label"
              title="February"
            >
              February
            </span>
            <span
              class="MicroBarChart-text"
              title="100"
            >
              100
            </span>
          </div>
          <div
            class="MicroBarChart-valueContainer"
            style="opacity: 1; height: 4px;"
          >
            <div
              class="MicroBarChart-valueBar"
              style="width: 100%;"
            />
          </div>
        </div>
        <div
          class="MicroBarChart-barContainer"
        >
          <div
            class="MicroBarChart-labelContainer"
          >
            <span
              class="MicroBarChart-label"
              title="March"
            >
              March
            </span>
            <span
              class="MicroBarChart-text"
              title="70"
            >
              70
            </span>
          </div>
          <div
            class="MicroBarChart-valueContainer"
            style="opacity: 1; height: 4px;"
          >
            <div
              class="MicroBarChart-valueBar"
              style="width: 70%;"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Micro Bar Chart With custom colors 1`] = `
<DocumentFragment>
  <div
    style="font-family: var(--sapFontFamily); width: 100%; height: 400px; position: relative;"
  >
    <div
      class="recharts-responsive-container"
      style="width: 400px; height: 400px;"
    >
      <div
        class="MicroBarChart-container"
        height="400"
        width="400"
      >
        <div
          class="MicroBarChart-barContainer"
        >
          <div
            class="MicroBarChart-labelContainer"
          >
            <span
              class="MicroBarChart-label"
              title="January - formatted"
            >
              January - formatted
            </span>
            <span
              class="MicroBarChart-text"
              title="10"
            >
              10
            </span>
          </div>
          <div
            class="MicroBarChart-valueContainer"
            style="opacity: 1; height: 4px;"
          >
            <div
              class="MicroBarChart-valueBar"
              style="width: 10%; background-color: black;"
            />
          </div>
        </div>
        <div
          class="MicroBarChart-barContainer"
        >
          <div
            class="MicroBarChart-labelContainer"
          >
            <span
              class="MicroBarChart-label"
              title="February - formatted"
            >
              February - formatted
            </span>
            <span
              class="MicroBarChart-text"
              title="100"
            >
              100
            </span>
          </div>
          <div
            class="MicroBarChart-valueContainer"
            style="opacity: 1; height: 4px;"
          >
            <div
              class="MicroBarChart-valueBar"
              style="width: 100%; background-color: yellow;"
            />
          </div>
        </div>
        <div
          class="MicroBarChart-barContainer"
        >
          <div
            class="MicroBarChart-labelContainer"
          >
            <span
              class="MicroBarChart-label"
              title="March - formatted"
            >
              March - formatted
            </span>
            <span
              class="MicroBarChart-text"
              title="70"
            >
              70
            </span>
          </div>
          <div
            class="MicroBarChart-valueContainer"
            style="opacity: 1; height: 4px;"
          >
            <div
              class="MicroBarChart-valueBar"
              style="width: 70%;"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
